### PR TITLE
Tfdoc features

### PR DESCRIPTION
**Pull request by Artlvns**
feat: inject .terraform-docs.yml config files into repo root.  
feat: inject .terraform-docs.yml files into examples folder if one exists and not configured.  
feat: generate tfdocs for examples if exist.

### CODE REVIEW

**Comment by Artlvns** [comment]
@Laffs2k5  here is initial commit with logic. I found a way to minimize bash script size and also added some logic that allow action to be used if examples do not exist.  I believe that it can be then potentially reused other places as well. 
Please check and come with feedback. I can also rewrite to separate .sh with functions if needed. Per now I leaved it as is since script is not complex nor large.

File: terraform-docs/action.yaml [hunk 1]
```diff
@@ -7,12 +7,45 @@ inputs:
...
+        set -o allexport; source "${{ github.action_path }}/helpers.sh"; set +o allexport;
+
+        # check if terraform-docs config is present in the repo root
+        if [ -f ".terraform-docs.yml" ]; then
+            log-info ".terraform-docs.yml found in the repo root. Nothing to do."
+        else
+            log-info ".terraform-docs.yml not found in the repo root. Creating default config."
+            cp "${{ github.action_path }}/terraform-docs-module-root.yml" "${{ github.workspace }}/.terraform-docs.yml"
+            log-info ".terraform-docs.yml created in the repo root."
+            ls -la
```
Comment (Laffs2k5) [inline terraform-docs/action.yaml:29]:
```suggestion
            log-multiline "root directory content" "$(ls -la)"
```

File: terraform-docs/action.yaml [hunk 2]
```diff
@@ -7,12 +7,45 @@ inputs:
...
+            log-info "examples folder found in the repo root."
+            set-output "examples-folder-exists" "true"
+            # check if .terraform-docs.yml config is present in the examples folder
+            if [ -f "examples/terraform-docs.yml" ]; then
+              log-info ".terraform-docs.yml found in the examples folder. Nothing to do."
+            else
+              log-info ".terraform-docs.yml not found in the examples folder. Creating default config."
+              cp "${{ github.action_path }}/terraform-docs-module-examples.yml" "${{ github.workspace }}/examples/.terraform-docs.yml"
+              log-info ".terraform-docs.yml created in the examples folder."
+            fi
```
Comment (Laffs2k5) [inline terraform-docs/action.yaml:43]:
```suggestion
            fi
            log-multiline "example directory content" "$(ls -la)"
```

File: terraform-docs/action.yaml [hunk 3]
```diff
@@ -7,12 +7,45 @@ inputs:
...
+        else
+            log-info "examples folder not found in the repo root. Examples docs will not be generated."
```
Comment (Laffs2k5) [inline terraform-docs/action.yaml:45]:
```suggestion
            log-info "examples folder not found in the repo root. Examples docs will not be generated."
            set-output "examples-folder-exists" "false"
```

File: terraform-docs/action.yaml [hunk 4]
```diff
@@ -50,16 +83,27 @@ runs:
...
         fi
       continue-on-error: true
-    - id: run-terraform-docs
-      if: steps.validate-readme.outcome == 'success'
+
+    - id: generate-example-docs
+      if: steps.inject-config-files.outputs.examples-folder-exists == 'true'
+      uses: terraform-docs/gh-actions@aeae0038ed47a547e0c0fca5c059d3335f48fb25
+      with:
+        output-method: inject
```
Comment (Laffs2k5) [inline terraform-docs/action.yaml:91]: double `inject`, both from action and in config. you do not need `inject` here, it's the default ;-)

ref. <https://terraform-docs.io/user-guide/configuration/output/>

File: terraform-docs/action.yaml [hunk 5]
```diff
@@ -50,16 +83,27 @@ runs:
...
+        recursive-path: "examples"
+        config-file: examples/.terraform-docs.yml
+
+    - id: generate-project-docs
+      if: steps.validate-root-readme.outcome == 'success'
       uses: terraform-docs/gh-actions@aeae0038ed47a547e0c0fca5c059d3335f48fb25
       with:
         working-dir: ${{ inputs.readme-file-path }}
-        output-file: README.md
         output-method: inject
```
Comment (Laffs2k5) [inline terraform-docs/action.yaml:102]: you do not need `inject` here, it's the default ;-)

ref. <https://terraform-docs.io/user-guide/configuration/output/>

File: terraform-docs/terraform-docs-module-examples.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  {{ .Header }}
+
+  ```hcl
+  {{ include "main.tf" }}
+  ```
+  {{ .Outputs }}
+
+output:
+  file: README.md
+  mode: inject
```
Comment (Laffs2k5) [inline terraform-docs/terraform-docs-module-examples.yml:24]: you do not need `inject` here, it's the default ;-)

ref. <https://terraform-docs.io/user-guide/configuration/output/>

**Review by Laffs2k5 — APPROVED** [review]
Nice work! Just a little polish and we're there 🤩

File: terraform-docs/action.yaml (hunk 4 above)
Comment (Artlvns) [reply terraform-docs/action.yaml:93]: Correct, but we need to set it as "empty string" if output.mode and output.file are defined in config file. Ref [action doc](https://github.com/terraform-docs/gh-actions?tab=readme-ov-file#inputs)

File: terraform-docs/action.yaml (hunk 5 above)
Comment (Artlvns) [reply terraform-docs/action.yaml:104]: same as above

**Pull request merged**